uzsu/__init__.py

```diff
@@ -679,25 +733,24 @@ def _schedule(self, item, caller=None):
                 _reset_interpolation = True
             elif _interpolation.lower() == 'cubic' and _interval > 0:
                 try:
-                    tck = interpolate.PchipInterpolator(list(self._itpl[item].keys()), list(self._itpl[item].values()))
                     _nextinterpolation = datetime.now(self._timezone) + timedelta(minutes=_interval)
                     _next = _nextinterpolation if _next > _nextinterpolation else _next
-                    _value = round(float(tck(_next.timestamp() * 1000.0)), self._interpolation_precision)
-                    _value_now = round(float(tck(entry_now)), self._interpolation_precision)
+                    _value = self._cubic(self._itpl[item], _next.timestamp() * 1000.0)
+                    _value_now = self._cubic(self._itpl[item], entry_now)
                     if _caller != "dry_run":
                         self._set(item=item, value=_value_now, caller=_caller)
                     self.logger.info("Updated: {}, cubic interpolation value: {}, based on dict: {}."
                                      " Next: {}, value: {}".format(item, _value_now, self._itpl[item], _next, _value))
+                    self.logger.info(f'New values: {self._cubic(self._itpl[item], entry_now)}, next: {self._cubic(self._itpl[item], _next.timestamp() * 1000.0)}')
```

Comment:
Could the two logging messages be merged (delete first, extend second)?

Reply:
No. The second was the control logger for the original and my interpolation methods. Will be removed :)